### ISSUE TITLE
Backend IPv6 in Consul Catalog

### ISSUE THREAD

**Issue by seeder** [containous/traefik#714]
We are using consul_catalog with IPv6 only services.
Traefik correctly discovers their addresses, however it uses malformed http config for backend.

Backends get constructed as `http://200X:XXX:XX:d3:0:242:ac11:4:8300`

instead of `http://[200X:XXX:XX:d3:0:242:ac11:4]:8300`

and causing errors: 

`Error forwarding to http://200X:XXX:XX:d3:0:242:ac11:5:8080, err: dial tcp: too many colons in address 200X:XXX:XX:d3:0:242:ac11:5:8080"`

**Comment by seeder** [comment]
So , my crude attempt at passing BackendAddress with, or without square brackets depending if it's IPv4 or IPv6 address is here:

https://github.com/seeder/traefik/tree/ipv6-consul-catalog

**Comment by traefiker** [comment]
I'll close the issue due to inactivity but feel free to re-open if needed.

**Comment by herver** [comment]
Hi there,

Just got hit by this one, it is very similar to #3428 and most likely the fix is identical.

### Generated backend config:
```json
   "radosgw-0-24j0duzKFiHUlSC0ABuYpAd8jic": {
     "url": "http://2001:xxx:xxx:e3::100:13d:8080",
     "weight": 1
   }
```

### Output of a test curl:
```
 curl --http1.1 -v -k -H 'Host: xxxx' http://cephxxxxx/
*   Trying 2001:xxx:xxx:e3::100:13d...
* TCP_NODELAY set
* Connected to cephxxxxx (2001:xxx:xxx:e3::100:13d) port 80 (#0)
> GET / HTTP/1.1
> Host: xxxx
> User-Agent: curl/7.60.0
> Accept: */*
> 
< HTTP/1.1 500 Internal Server Error
< Date: Tue, 12 Jun 2018 08:39:57 GMT
< Content-Length: 21
< Content-Type: text/plain; charset=utf-8
< 
* Connection #0 to host cephxxxxx left intact
```

### Desired backend config:
```json
   "radosgw-0-24j0duzKFiHUlSC0ABuYpAd8jic": {
     "url": "http://[2001:xxx:xxx:e3::100:13d]:8080",
     "weight": 1
   }
```

PR incoming...

**Comment by traefiker** [comment]
Closed by #3477.
